### PR TITLE
Partial cleanup of GPU feature detection

### PR DESCRIPTION
**Pull request by hrydgard**
This is probably almost as far as we get without using better GL headers (that has everything whether we compile for Android or Windows or whatever).

The end goal is to be able to run real OpenGL on Android (which is supported on most Nvidia devices), and also OpenGL ES on Windows for testing. Currently neither of these are possible.

Corresponding changes need to be done to native: https://github.com/hrydgard/native/pull/287

### CODE REVIEW

File: UI/EmuScreen.cpp [hunk 1]
```diff
@@ -180,6 +180,7 @@ void EmuScreen::dialogFinished(const Screen *dialog, DialogResult result) {
 		quit_ = false;
 	}
 	RecreateViews();
+	gpu->CheckGPUFeatures();
```
Comment (unknownbrackets) [inline UI/EmuScreen.cpp]: Wouldn't this go better e.g. one of these places where we do similar things?

https://github.com/hrydgard/ppsspp/blob/gpu-feature-cleanup/GPU/GLES/GLES_GPU.cpp#L683
https://github.com/hrydgard/ppsspp/blob/gpu-feature-cleanup/GPU/GLES/GLES_GPU.cpp#L2277

Other places send signals that call Resized when changing settings.  I feel like it's better to centralize.

-[Unknown]

Comment (hrydgard) [inline UI/EmuScreen.cpp]: Yeah, done.

**Comment by unknownbrackets** [comment]
The only thing I have to add is #7943.  LGTM, let's merge.

-[Unknown]

File: GPU/GLES/GLES_GPU.cpp [hunk 2]
```diff
@@ -463,6 +466,94 @@ GLES_GPU::~GLES_GPU() {
...
+			features |= GPU_SUPPORTS_GLSL_330;
+	}
+
+	// Framebuffer fetch appears to be buggy at least on Tegra 3 devices.  So we blacklist it.
+	// Tales of Destiny 2 has been reported to display green.
+	if (gl_extensions.EXT_shader_framebuffer_fetch || gl_extensions.NV_shader_framebuffer_fetch || gl_extensions.ARM_shader_framebuffer_fetch) {
+		features |= GPU_SUPPORTS_ANY_FRAMEBUFFER_FETCH;
+		// Blacklist Tegra 3, doesn't work very well.
+		if (strstr(gl_extensions.model, "NVIDIA Tegra 3") != 0) {
+			features &= ~GPU_SUPPORTS_ANY_FRAMEBUFFER_FETCH;
```
Comment (sum2012) [inline GPU/GLES/GLES_GPU.cpp]: @hrydgard So here is for blacklist for my phone and @shinemu phone's code (PowerVR SGX 540) ?

Comment (hrydgard) [inline GPU/GLES/GLES_GPU.cpp]: Yes, this is the correct place to add those checks.

Comment (sum2012) [inline GPU/GLES/GLES_GPU.cpp]: Thanks

File: UI/EmuScreen.cpp [hunk 3]
```diff
@@ -923,20 +923,20 @@ void EmuScreen::render() {
 		screenManager()->getUIContext()->End();
 	}
 
-#ifdef USING_GLES2
 	// We have no use for backbuffer depth or stencil, so let tiled renderers discard them after tiling.
 	if (gl_extensions.GLES3 && glInvalidateFramebuffer != nullptr) {
```
Comment (unknownbrackets) [inline UI/EmuScreen.cpp]: FWIW, I think some of our Linux builds relied on this ifdef due to glew.  Need a better loader...

-[Unknown]

**Pull request merged**